File: backend/backend/models/chapter/chapter.js
const mongoose = require("mongoose");

const chapterSchema = mongoose.Schema(
  {
    avatar: {
      type: String,
      required: true,
    },
    name: {
      type: String,
      required: true,
      unique: true,
    },
    title: {
      type: String,
      required: true,
      unique: true,
    },
    isCompleted: {
      type: Boolean,
      required: true,
      default: false,
    },
    isUnlocked: {
      type: Boolean,
      required: true,
      default: false,
    },
    color: {
      type: String,
      required: true,
    },
    courseId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Course",
    },
    sectionIds: [
      {
        type: mongoose.Schema.Types.ObjectId,
        ref: "Section",
      },
    ],
  },
  { timestamp: true }
);

// export
const Chapter = mongoose.model("Chapter", chapterSchema);
module.exports = Chapter;

File: frontend/doulingo/lib/domain/chapter/entities/chapter.dart
class ChapterEntity {
  final String? id;
  final String? avatar;
  final String? name;
  final String? title;
  final bool? isCompleted;
  final bool? isUnlocked;
  final String? color;
  final dynamic courseId;
  final List<dynamic>? sectionIds;

  ChapterEntity({
    this.id,
    this.avatar,
    this.name,
    this.title,
    this.isCompleted,
    this.isUnlocked,
    this.color,
    this.courseId,
    this.sectionIds,
  });
}
